src/main/java/com/gizasystems/automationexercise/pages/NavigationBar.java

```diff
@@ -31,10 +33,24 @@ public NavigationBar clickOnDeleteAccountLink() {
         return this;
     }
 
+    @Step("Click on Products link")
+    public NavigationBar clickOnProductsLink() {
+        driver.element().click(products);
+        return this;
+    }
+    @Step("Handle ADS")
+    public NavigationBar handleAd(){
+        Actions actions = new Actions(driver.getDriver());
+        actions.pause(1000).doubleClick().perform();
+        return this;
+    }
```

Comment:
Why handling the add in the navigation bar page?
Also, I believe we already have a method for this (please check the utils package).